### PR TITLE
Align fold, reduce, scan ops #60

### PR DESCRIPTION
**Pull request by jpork**

### CODE REVIEW

File: src/main/scala/reactor/core/scala/publisher/SFluxLike.scala [hunk 1]
```diff
@@ -71,13 +75,11 @@ trait SFluxLike[+T] extends ScalaConverters {
     coreFlux.onErrorResume(f).asScala
   }
 
-  final def reduce[A](initial: A)(accumulator: (A, T) => A): SMono[A] = coreFlux.reduce[A](initial, accumulator).asScala
```
Comment (sinwe) [inline src/main/scala/reactor/core/scala/publisher/SFluxLike.scala:74]: So we are removing `reduce` and people should use `fold`?
This is breaking changes. Shouldn't we at least make it as deprecated first?

File: src/main/scala/reactor/core/scala/publisher/SFluxLike.scala [hunk 2]
```diff
@@ -36,18 +36,22 @@ trait SFluxLike[+T] extends ScalaConverters {
...
   final def flatten[S](implicit ev: T <:< SFlux[S]): SFlux[S] = concatMap[S](x => ev(x), XS_BUFFER_SIZE)
 
-  final def fold[R](initial: R)(binaryOps: (R, T) => R): SMono[R] = reduce(initial)(binaryOps)
+  final def fold[R](initial: R)(binaryOps: (R, T) => R): SMono[R] = {
+    coreFlux.reduce(initial, binaryOps).asScala
+  }
 
-  final def foldLeft[R](initial: R)(binaryOps: (R, T) => R): SMono[R] = reduce[R](initial)(binaryOps)
+  final def foldWith[R](initial: => R)(binaryOps: (R, T) => R): SMono[R] = {
+    coreFlux.reduceWith(()=> initial, binaryOps).asScala
```
Comment (sinwe) [inline src/main/scala/reactor/core/scala/publisher/SFluxLike.scala:44]: Nice catch, I missed this

File: src/main/scala/reactor/core/scala/publisher/SFluxLike.scala (hunk 1 above)
Comment (jpork) [reply src/main/scala/reactor/core/scala/publisher/SFluxLike.scala:74]: yup. forward to fold and making it deprecate would be great. shall I do so?

**Comment by jpork** [comment]
Did you cross-check with scala api doc just to ensure if this change is ok at all?
I've consulted also the [rxscala api](http://reactivex.io/rxscala/scaladoc/index.html#rx.lang.scala.Observable). 
They have foldLeft and reduce (without left).

File: src/main/scala/reactor/core/scala/publisher/SFluxLike.scala (hunk 1 above)
Comment (sinwe) [reply src/main/scala/reactor/core/scala/publisher/SFluxLike.scala:74]: @jpork Yes please. I think that'd be prudent to do.

**Comment by sinwe** [comment]
> Did you cross-check with scala api doc just to ensure if this change is ok at all?
> I've consulted also the [rxscala api](http://reactivex.io/rxscala/scaladoc/index.html#rx.lang.scala.Observable).
> They have foldLeft and reduce (without left).

They don't maintain rxscala for the rxjava 2 anymore. I don't even know if they have plan for rxscala for rxjava3

File: src/main/scala/reactor/core/scala/publisher/SFluxLike.scala (hunk 1 above)
Comment (sinwe) [reply src/main/scala/reactor/core/scala/publisher/SFluxLike.scala:74]: @jpork I just realize that we don't have `reduce` without initial value here.

Comment (jpork) [reply src/main/scala/reactor/core/scala/publisher/SFluxLike.scala:74]: Done.

**Comment by sinwe** [comment]
@jpork, I've approved, but apparently there is conflicts. Can you resolve the conflict?

**Comment by jpork** [comment]
I've resolved the upstream conflicts.

**Pull request merged**